File: MiniDOMOffice/ADVs/Campagne/CCampagnaPubblicitariaCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class ADV
    {

        /// <summary>
        /// Cursore sulla tabella <see cref="CCampagnaPubblicitaria"/>
        /// </summary>
        [Serializable]
        public class CCampagnaPubblicitariaCursor 
            : minidom.Databases.DBObjectCursor<CCampagnaPubblicitaria>
        {
            private DBCursorStringField m_NomeCampagna = new DBCursorStringField("NomeCampagna");
            private DBCursorStringField m_Titolo = new DBCursorStringField("Titolo");
            private DBCursorStringField m_Testo = new DBCursorStringField("Testo");
            // Private m_TipoTesto As New DBCursorFieldObj(Of String)("TipoTesto")
            private DBCursorField<bool> m_UsaListaDinamica = new DBCursorField<bool>("UsaListaDinamica");
            private DBCursorStringField m_ParametriLista = new DBCursorStringField("ParametriLista");
            private DBCursorField<int> m_IDListaDestinatari = new DBCursorField<int>("IDListaDestinatari");
            private DBCursorField<bool> m_Attiva = new DBCursorField<bool>("Attiva");
            private DBCursorStringField m_NomeMittente = new DBCursorStringField("NomeMittente");
            private DBCursorStringField m_IndirizzoMittente = new DBCursorStringField("IndirizzoMittente");
            private DBCursorField<StatoCampagnaPubblicitaria> m_StatoCampagna = new DBCursorField<StatoCampagnaPubblicitaria>("StatoCampagna");
            private DBCursorField<bool> m_RichiediConfermaDiLettura = new DBCursorField<bool>("ConfermaLettura");
            private DBCursorField<bool> m_RichiediConfermaDiRecapito = new DBCursorField<bool>("ConfermaRecapito");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");
            private DBCursorStringField m_FileDaUtilizzare = new DBCursorStringField("FileDaUtilizzare");
            private DBCursorStringField m_ListaCC = new DBCursorStringField("ListaCC");
            private DBCursorStringField m_ListaCCN = new DBCursorStringField("ListaCCN");
            private DBCursorStringField m_ListaNO = new DBCursorStringField("ListaNO");

            /// <summary>
            /// Costruttore
            /// </summary>
            public CCampagnaPubblicitariaCursor()
            {
            }

            /// <summary>
            /// ListaNO
            /// </summary>
            public DBCursorStringField ListaNO
            {
                get
                {
                    return m_ListaNO;
                }
            }

            /// <summary>
            /// ListaCC
            /// </summary>
            public DBCursorStringField ListaCC
            {
                get
                {
                    return m_ListaCC;
                }
            }

            /// <summary>
            /// ListaCCN
            /// </summary>
            public DBCursorStringField ListaCCN
            {
                get
                {
                    return m_ListaCCN;
                }
            }

            /// <summary>
            /// FileDaUtilizzare
            /// </summary>
            public DBCursorStringField FileDaUtilizzare
            {
                get
                {
                    return m_FileDaUtilizzare;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<int> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// NomeCampagna
            /// </summary>
            public DBCursorStringField NomeCampagna
            {
                get
                {
                    return m_NomeCampagna;
                }
            }

            /// <summary>
            /// Titolo
            /// </summary>
            public DBCursorStringField Titolo
            {
                get
                {
                    return m_Titolo;
                }
            }

            /// <summary>
            /// Testo
            /// </summary>
            public DBCursorStringField Testo
            {
                get
                {
                    return m_Testo;
                }
            }

            /// <summary>
            /// UsaListaDinamica
            /// </summary>
            public DBCursorField<bool> UsaListaDinamica
            {
                get
                {
                    return m_UsaListaDinamica;
                }
            }

            /// <summary>
            /// ParametriLista
            /// </summary>
            public DBCursorStringField ParametriLista
            {
                get
                {
                    return m_ParametriLista;
                }
            }

            /// <summary>
            /// IDListaDestinatari
            /// </summary>
            public DBCursorField<int> IDListaDestinatari
            {
                get
                {
                    return m_IDListaDestinatari;
                }
            }

            /// <summary>
            /// Attiva
            /// </summary>
            public DBCursorField<bool> Attiva
            {
                get
                {
                    return m_Attiva;
                }
            }

            /// <summary>
            /// NomeMittente
            /// </summary>
            public DBCursorStringField NomeMittente
            {
                get
                {
                    return m_NomeMittente;
                }
            }

            /// <summary>
            /// IndirizzoMittente
            /// </summary>
            public DBCursorStringField IndirizzoMittente
            {
                get
                {
                    return m_IndirizzoMittente;
                }
            }

            /// <summary>
            /// StatoCampagna
            /// </summary>
            public DBCursorField<StatoCampagnaPubblicitaria> StatoCampagna
            {
                get
                {
                    return m_StatoCampagna;
                }
            }

            /// <summary>
            /// RichiediConfermaDiLettura
            /// </summary>
            public DBCursorField<bool> RichiediConfermaDiLettura
            {
                get
                {
                    return m_RichiediConfermaDiLettura;
                }
            }

            /// <summary>
            /// RichiediConfermaDiRecapito
            /// </summary>
            public DBCursorField<bool> RichiediConfermaDiRecapito
            {
                get
                {
                    return m_RichiediConfermaDiRecapito;
                }
            }

            // Public ReadOnly Property TipoTesto As DBCursorFieldObj(Of String)
            // Get
            // Return Me.m_TipoTesto
            // End Get
            // End Property

             
            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.ADV.Campagne;
            }

            
        }
    }
}

File: MiniDOMOffice/ADVs/CCampagnaPubblicitariaCursor.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Anagrafica
Imports minidom.CustomerCalls

Imports minidom.Sistema
Imports System.Net.Mail

Partial Public Class ADV
 

    Public Class CCampagnaPubblicitariaCursor
        Inherits DBObjectCursor(Of CCampagnaPubblicitaria)

        Private m_NomeCampagna As New CCursorFieldObj(Of String)("NomeCampagna")
        Private m_Titolo As New CCursorFieldObj(Of String)("Titolo")
        Private m_Testo As New CCursorFieldObj(Of String)("Testo")
        'Private m_TipoTesto As New CCursorFieldObj(Of String)("TipoTesto")
        Private m_UsaListaDinamica As New CCursorField(Of Boolean)("UsaListaDinamica")
        Private m_ParametriLista As New CCursorFieldObj(Of String)("ParametriLista")
        Private m_IDListaDestinatari As New CCursorField(Of Integer)("IDListaDestinatari")
        Private m_Attiva As New CCursorField(Of Boolean)("Attiva")
        Private m_NomeMittente As New CCursorFieldObj(Of String)("NomeMittente")
        Private m_IndirizzoMittente As New CCursorFieldObj(Of String)("IndirizzoMittente")
        Private m_StatoCampagna As New CCursorField(Of StatoCampagnaPubblicitaria)("StatoCampagna")
        Private m_RichiediConfermaDiLettura As New CCursorField(Of Boolean)("ConfermaLettura")
        Private m_RichiediConfermaDiRecapito As New CCursorField(Of Boolean)("ConfermaRecapito")
        Private m_Flags As New CCursorField(Of Integer)("Flags")
        Private m_FileDaUtilizzare As New CCursorFieldObj(Of String)("FileDaUtilizzare")
        Private m_ListaCC As New CCursorFieldObj(Of String)("ListaCC")
        Private m_ListaCCN As New CCursorFieldObj(Of String)("ListaCCN")
        Private m_ListaNO As New CCursorFieldObj(Of String)("ListaNO")

        Public Sub New()
        End Sub

        Public ReadOnly Property ListaNO As CCursorFieldObj(Of String)
            Get
                Return Me.m_ListaNO
            End Get
        End Property

        Public ReadOnly Property ListaCC As CCursorFieldObj(Of String)
            Get
                Return Me.m_ListaCC
            End Get
        End Property

        Public ReadOnly Property ListaCCN As CCursorFieldObj(Of String)
            Get
                Return Me.m_ListaCCN
            End Get
        End Property

        Public ReadOnly Property FileDaUtilizzare As CCursorFieldObj(Of String)
            Get
                Return Me.m_FileDaUtilizzare
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property NomeCampagna As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCampagna
            End Get
        End Property

        Public ReadOnly Property Titolo As CCursorFieldObj(Of String)
            Get
                Return Me.m_Titolo
            End Get
        End Property

        Public ReadOnly Property Testo As CCursorFieldObj(Of String)
            Get
                Return Me.m_Testo
            End Get
        End Property

        Public ReadOnly Property UsaListaDinamica As CCursorField(Of Boolean)
            Get
                Return Me.m_UsaListaDinamica
            End Get
        End Property

        Public ReadOnly Property ParametriLista As CCursorFieldObj(Of String)
            Get
                Return Me.m_ParametriLista
            End Get
        End Property

        Public ReadOnly Property IDListaDestinatari As CCursorField(Of Integer)
            Get
                Return Me.m_IDListaDestinatari
            End Get
        End Property

        Public ReadOnly Property Attiva As CCursorField(Of Boolean)
            Get
                Return Me.m_Attiva
            End Get
        End Property

        Public ReadOnly Property NomeMittente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeMittente
            End Get
        End Property

        Public ReadOnly Property IndirizzoMittente As CCursorFieldObj(Of String)
            Get
                Return Me.m_IndirizzoMittente
            End Get
        End Property

        Public ReadOnly Property StatoCampagna As CCursorField(Of StatoCampagnaPubblicitaria)
            Get
                Return Me.m_StatoCampagna
            End Get
        End Property

        Public ReadOnly Property RichiediConfermaDiLettura As CCursorField(Of Boolean)
            Get
                Return Me.m_RichiediConfermaDiLettura
            End Get
        End Property

        Public ReadOnly Property RichiediConfermaDiRecapito As CCursorField(Of Boolean)
            Get
                Return Me.m_RichiediConfermaDiRecapito
            End Get
        End Property

        'Public ReadOnly Property TipoTesto As CCursorFieldObj(Of String)
        '    Get
        '        Return Me.m_TipoTesto
        '    End Get
        'End Property


        Protected Overrides Function GetConnection() As CDBConnection
            Return CRM.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return ADV.Campagne.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_ADVs"
        End Function

    End Class


End Class
